### ISSUE TITLE
Feature request: make 'fields with numbers' represent real PIDs on the system, having them colored depending on high CPU/MemUsage

### ISSUE THREAD

**Issue by SjoerdV** [lcarsde/status-bar#6]
First off: very cool project, dear sir!

Wouldn't it be cool to make this part of the dashboard (fields with numbers) actually mean something useful by showing some real PIDs? 

Surely not in real-time, but with 5 or 10 second intervals or so?

Creating some color scale for the field buttons to denote the load a process is consuming would be nice too.

Furthermore maybe we could even split this section into 'user' and 'system' PIDs, or have distinctive feature (like colored border) for the fields to differentiate between 'user' and 'system' PIDs?

Professional UX/UI artists may have some nice ideas on this front, I dare ask 🫣

**Comment by kluzny** [comment]
PID - Binary Basename would be more useful IMHO, because PIDs are pretty random.

**Comment by atennert** [comment]
Yes, it looks pretty random. In my opinion that's part of it, because when we look at the screens in the series there are often these fields with numbers that we as viewers have no clue what it is. (At least I didn't :sweat_smile:)

That said, I'm open to using the names too. In the end,it would probably make sense to make it configurable via the `status-config.xml`. That way it's possible to choose between random numbers, PIDs, binary basenames and maybe other future options.

**Comment by techy2493** [comment]
I really like this idea. Something to make it have information.

Even though in the displays we have had of LCARS they appear random to us a system like this wouldn't have meaningless data presented. Having some sort of useful data makes sense. 

I don't think new colors should be introduced. The system already has 7 colors and that feels right. 

If you wanted to make the filler data represent system processes that were meaningful to the user with diagnostic information present what about...

A configurable dictionary of user defined values for processes they care about for numbers or letters.  Display everything else as PIDs.
Feature Creep: Allow clicking one to display detailed info and diagnostics for that process in place of the filler bar.
A color scheme using the existing colors higher values being represented by brighter colors.
Use the filler calculations that already exist for number of processes to be shown.
Use 5 colors, very low, low, mid, high, and very high. This could be based on the % of usage of total or just those displayed. 
Use a 6th color to represent a process in a fault state i.e. that is not responding or at a user defined threshold of total system resources.
Allow the user to choose 4 modes. Show CPU, Show Memory, Show both (using a left or right side end cap with user choosing one memory or cpu as primary), Show both when secondary value is above threshold of total system resources.
Feature Creep: Allow the user to define filters to only show processes they care about.
Feature Creep: Gold Plating: make filters contextual.

**Comment by atennert** [comment]
Thank you for the input. No guarantee for any timeline at the moment though, sorry.
